File: skills/meta-analysis/tests/run_all.R
#!/usr/bin/env Rscript
# Master validation script for the meta-analysis skill.
#
# Runs against datasets shipped with metadat and netmeta. Nothing is downloaded,
# so the whole suite completes in under a minute offline.
#
# Usage:
#   Rscript run_all.R              # run all tests
#   Rscript run_all.R effects      # run one test
#
# Expected runtime: under 1 minute
#
# Requirements:
#   metafor, metadat   effects, heterogeneity, bias
#   netmeta, meta      network meta-analysis

args <- commandArgs(trailingOnly = TRUE)

tests <- c("effects", "heterogeneity", "bias", "nma")
if (length(args) > 0) {
  tests <- match.arg(args[1], tests)
}

cat("Meta-Analysis Skill Validation\n")
cat("===============================\n")
cat(sprintf("Date: %s\n", Sys.time()))
cat(sprintf("R version: %s\n", R.version.string))

for (p in c("metafor", "metadat", "netmeta", "meta")) {
  v <- tryCatch(as.character(packageVersion(p)), error = function(e) "NOT INSTALLED")
  cat(sprintf("%s: %s\n", p, v))
}
cat(sprintf("Tests to run: %s\n\n", paste(tests, collapse = ", ")))

script_dir <- tryCatch(dirname(sys.frame(1)$ofile), error = function(e) ".")
if (is.null(script_dir) || is.na(script_dir) || script_dir == "") script_dir <- "."

failed <- character()

for (test in tests) {
  script <- file.path(script_dir, sprintf("validate_%s.R", test))
  if (file.exists(script)) {
    cat(sprintf("\n--- Running %s validation ---\n\n", test))
    status <- system2("Rscript", script)
    if (!identical(status, 0L)) failed <- c(failed, test)
  } else {
    cat(sprintf("Script not found: %s\n", script))
    failed <- c(failed, test)
  }
}

cat("\n===============================\n")
if (length(failed) == 0) {
  cat("All validations passed.\n")
} else {
  cat(sprintf("Failed: %s\n", paste(failed, collapse = ", ")))
  quit(status = 1)
}

File: skills/epigenomics/tests/run_all.py
#!/usr/bin/env python3
"""Master validation script for the epigenomics skill.

Usage:
    python run_all.py            # all suites
    python run_all.py claims

Suites skip cleanly (exit 0) when a dependency or network is unavailable, so
a partial environment reports what it could not check rather than failing.
"""
import subprocess
import sys
from pathlib import Path

SUITES = ["claims"]

args = sys.argv[1:]
suites = args or SUITES
unknown = [a for a in suites if a not in SUITES]
if unknown:
    print(f"Unknown suite(s): {unknown}. Choose from {SUITES}")
    sys.exit(2)

here = Path(__file__).resolve().parent
print("epigenomics validation")
print("=" * 40)
print(f"Python: {sys.version.split()[0]}")
print(f"Suites: {', '.join(suites)}\n")

failed = []
for s in suites:
    script = here / f"validate_{s}.py"
    if not script.exists():
        print(f"Script not found: {script}"); failed.append(s); continue
    print(f"\n--- Running {s} validation ---\n")
    if subprocess.run([sys.executable, str(script)]).returncode != 0:
        failed.append(s)

print("\n" + "=" * 40)
if failed:
    print(f"Failed: {', '.join(failed)}"); sys.exit(1)
print("All validations passed.")
